libfreerdp/core/orders.c

```diff
@@ -1153,11 +1171,12 @@ static INLINE BOOL FIELD_SKIP_BUFFER16(wStream* s, UINT32 TARGET_LEN)
 /* Primary Drawing Orders */
 static BOOL update_read_dstblt_order(wStream* s, const ORDER_INFO* orderInfo, DSTBLT_ORDER* dstblt)
 {
-	ORDER_FIELD_COORD(1, dstblt->nLeftRect);
-	ORDER_FIELD_COORD(2, dstblt->nTopRect);
-	ORDER_FIELD_COORD(3, dstblt->nWidth);
-	ORDER_FIELD_COORD(4, dstblt->nHeight);
-	ORDER_FIELD_BYTE(5, dstblt->bRop);
+	if (!read_order_field_coord(orderInfo, s, 1, &dstblt->nLeftRect, FALSE) ||
+	    !read_order_field_coord(orderInfo, s, 2, &dstblt->nTopRect, FALSE) ||
+	    !read_order_field_coord(orderInfo, s, 3, &dstblt->nWidth, FALSE) ||
+	    !read_order_field_coord(orderInfo, s, 4, &dstblt->nHeight, FALSE) ||
+	    !read_order_field_byte(orderInfo, s, 5, &dstblt->bRop, TRUE))
+		return FALSE;
 	return TRUE;
```

Comment:
So I guess that could be replaced by 
```C
return read_order_field_coord(orderInfo, s, 1, &dstblt->nLeftRect, FALSE) &&
 read_order_field_coord(orderInfo, s, 2, &dstblt->nTopRect, FALSE) &&
...
```

Reply:
true, but I prefer to be able to set a breakpoint.

Reply:
I think @hardening's main point is that `!a || !b || ... || !z` can be replaced with `a && b && ... && z` which makes the statement clearer